tests/test_gitdepotoperations.py

```diff
@@ -172,3 +184,73 @@ def test_request_refresh_git(self):
 
         self.assertTrue(workspace1.request_refresh({
             os.path.join(self.environment_path, 'remote'): ['my-branch1']}))
+
+    def test_request_refresh_git_dirty_workspace(self):
+        dcvs = DepotOperations()
+
+        # Remote repository
+        self.add_content_to_repo(
+            os.path.join(FIXTURE_PATH, 'fixture-2.git.bundle'),
+            'remote')
+
+        # Master cache
+        master = dcvs.init_depot(
+            os.path.join(self.environment_path, 'master'),
+            parent=None,
+            source=os.path.join(self.environment_path, 'remote'))
+
+        # Workspace depot
+        workspace1 = dcvs.init_depot(
+            os.path.join(self.environment_path, 'workspace1'),
+            parent=master,
+            source=os.path.join(self.environment_path, 'master'))
+
+        self.assertTrue(workspace1.request_refresh({
+            os.path.join(self.environment_path, 'remote'): ['my-branch1']}))
+
+        # Detach and taint the working copy
+        sh.git('checkout', detach=True, _cwd=workspace1.path)
+        f = open(os.path.join(workspace1.path, 'test1.txt'), 'w')
```

Comment:
When this file is written, would it be an untracked one ?
Do you think would it be interesting to test two scenarios here ? Adding untracked files and modifying files in the working copy ?

Reply:
No, this file exists, so it is seen as modified here. I can also try to add an untracked file, and ok, I can do a test for each scenario.

Reply:
I added four different scenarios for "dirty" working copies: modified file, untracked file, detached head, and everything together.